app/views/products/_form.html.erb

```diff
@@ -0,0 +1,8 @@
+<%= form_for [ @vendor, @product ] do |f| %>
+  <%= f.label :name, "Name:" %>
+  <%= f.text_field :name, required: :true %>
+
+  <%= f.hidden_field :vendor_id, value: params[:vendor_id] %>
```

Comment:
I don't know if I for sure remember the issue we had here, but I think it's because we'd talked about shallow nesting :products under :vendor in config/routes? Either way, whatever the reason, you're completely right.